vmdb/app/models/mixins/naming_index_mixin.rb

```diff
@@ -0,0 +1,20 @@
+module NamingIndexMixin
+  extend ActiveSupport::Concern
+
+  NAMING = 'naming'
+
+  included do
+    has_many :naming_indexes, :as => :resource, :dependent => :destroy, :class_name => "CustomAttribute", :conditions => {:source => NAMING}
+  end
+
+  def next_naming_index(name)
+    with_lock do
```

Comment:
Ours seems to be pretty much the same as `with_lock`, only with a different spelling to request a shared lock, and a built-in timeout. (Also, a lying debug message, but that's neither here nor there.)

What's the history of the timeout? The database should deal with deadlocks itself, so do we have situations where we do other potentially-slow things while holding a lock? Things that can be safely/usefully aborted? :confused:

AFAIK, AR instances don't have a `lock` method, so while we may be semantically conflicting, I don't think we have an actual issue.